internal/controller/apiextensions/claim/api.go

```diff
@@ -64,7 +64,10 @@ func (a *APIBinder) Bind(ctx context.Context, cm resource.CompositeClaim, cp res
 	}
 
 	// Propagate the actual external name back from the composite to the claim.
```

Comment:
It may be worth expanding on this comment to mention the fact that we do this conditionally in order to allow the claim's external name to be authoritative in cases where the XR does not yet have one.